### PR TITLE
Tests: improve coverage of NodeScanner

### PR DESCRIPTION
**Pull request by erlend-aasland**

### CODE REVIEW

File: test/test_network.py [hunk 1]
```diff
@@ -92,13 +92,81 @@ def hook(_, data, ts):
...
-        scanner = canopen.network.NodeScanner()
-        scanner.on_message_received(0x586)
-        scanner.on_message_received(0x587)
-        scanner.on_message_received(0x586)
-        self.assertListEqual(scanner.nodes, [6, 7])
+    def setUp(self):
+        self.scanner = canopen.network.NodeScanner()
+
+    def test_scanner_on_message_received(self):
+        self.scanner.on_message_received(0x081)
```
Comment (erlend-aasland) [inline test/test_network.py:101]: Remove duplicate line.

```suggestion
```

**Review by acolomb — APPROVED** [review]
Looked at the code on a high level, and it seems to do sensible things.  Let's get this merged and take care of any fallout when it occurs (none expected though).

**Comment by acolomb** [comment]
Seems flaky... It failed immediately when testing the merge commit on master.

@erlend-aasland could you please investigate?

**Comment by erlend-aasland** [comment]
> Seems flaky... [..] could you please investigate?

Sure; I'll get to it in a day or two.

**Comment by erlend-aasland** [comment]
`test_scanner_search` assumes all packets arrive sequentially, however that does not seem to be guaranteed by `python-can`. There is no need to check the order of the packages; we only need to check that all of them are sent. I'll rewrite the test to drop the sequential aspect.

**Pull request merged**